api-reference/api-introduction/api-keys.md

```diff
@@ -18,15 +34,44 @@ This should be kept confidential and only stored on your servers. Private key is
 
-1. Do not embed API keys directly in code. API keys that are embedded in code can be accidentally exposed to the public. For example, you may forget to remove the keys from code that you share. Instead of embedding your API keys in your applications, store them in environment variables or in files outside of your application's source tree.
+1. Do not embed API keys directly in your code. API keys that are embedded in code can be accidentally exposed to the public. For example, you may forget to remove the keys from the code that you share. Instead of embedding your API keys in your applications, store them in environment variables or in files outside of your application's source tree.
 2. Do not store API keys in files inside your application's source tree. If you store API keys in files, keep the files outside your application's source tree to help ensure your keys do not end up in your source code control system. This is particularly important if you use a public source code management system such as GitHub.
 
 ## Rolling keys
 
-If an API key is compromised, you should roll that pair and start using the newly generated keys.
+If your API key is compromised, you need to revoke or "roll" the key to block any API requests that might use that key. Doing this will revoke your API key and generate a new one with the same resource access permissions as the old one.
+
+You can choose when to expire the existing key:
+
+- Immediately
+- In 1 hour
+- In 24 hours
+- In 3 days
+- In 7 days
+
+The expiry period you choose blocks and expires the existing key after the time period you specify. Regardless of the expiry period, you can use the new key immediately.
+
+![Roll API keys](../../.gitbook/assets/roll-keys.png)
+
+## Deleting keys
+
+You can delete any existing API keys in your account. However, your account will have at least one standard key pair at all times.
+
+## Revealing keys
+
+By default, the private key is masked for security reasons. You can click on the reveal icon next to the private key and enter your password to authorise and reveal the API private keys.
+
+## Restricted API keys
+
+You can:
+
+- Create a new restricted API key and specify its resource access permissions.
+- Update resource access permissions on any existing restricted API keys.
+
+![Create restricted API keys](../../.gitbook/assets/create-new-restricted-key.png)
 
-![API keys in ImageKit.io dashboard](../../.gitbook/assets/rolling-keys.jpg)
+The three access levels across any resource are: `None`, `Read only` and `Read and write`.
 
-You can select an expiry time for the old key. After this duration the key pair will stop working.
+### Resource list
 
-![Roll keys from ImageKit.io dashboard](../../.gitbook/assets/roll-keys-expire-setting.jpg)
+- Media access: Permissions for operations on Upload, Media management, File metadata, and Custom metadata management APIs.
```

Comment:
Expand this into a table listing the different endpoints covered under each access level incrementally.

Reply:
Done